xwiki-platform-core/xwiki-platform-watchlist/xwiki-platform-watchlist-api/src/main/java/org/xwiki/watchlist/internal/RealtimeNotificationGenerator.java

```diff
@@ -115,17 +128,35 @@
     private ConfigurationSource xwikiProperties;
 
     /**
+     * Allow processing of remote events.
+     */
+    private boolean allowRemote;
+
+    /**
      * Default constructor.
      */
     public RealtimeNotificationGenerator()
     {
         super(LISTENER_NAME, EVENTS);
+        allowRemote = false;
+    }
+
+    /**
+     * Component manager initialize class. Called after all dependencies are injected.
+     *
+     * @throws InitializationException if component isn't properly initialized
+     */
+    public void initialize() throws InitializationException
+    {
+        if (TRUE.equals(xwikiProperties.getProperty("watchlist.realtime.allow_remote"))) {
```

Comment:
I'd have also preferred not to introduce a TRUE static final variable when Boolean.valueOf(), see https://docs.oracle.com/javase/7/docs/api/java/lang/Boolean.html#valueOf(java.lang.String)